### PR TITLE
Tooltip: refine existing tests

### PR DESCRIPTION
**Pull request by brookewp**
## What?

Refining the existing tooltip tests (and likely adding a few more but this is still a WIP)

## Why?

In preparation for a new tooltip component #48222, I wanted to refine and improve the existing tests for the legacy tooltip. As well as add any additional tests that are missing / would be beneficial. 

## How?

- By replacing `focus()` with `userEvent.tab()`to closer mimic user behaviour 
- Removing `act()` and fake timers 
- Replacing the button with our Button component (initially done because of [this issue](https://github.com/WordPress/gutenberg/issues/43129) but it looks like it might be related to the Icon / possibly other components. Still needs to be researched further)
- Refining the shortcut test to include what the prop is for

## Testing Instructions

Run `npm run test:unit packages/components/src/tooltip/test/index.js` and ensure tests are passing

### CODE REVIEW

File: packages/components/src/tooltip/test/index.js [hunk 1]
```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
...
-			expect( tooltip ).toBeVisible();
-			expect( mockOnFocus ).toHaveBeenCalledWith(
-				expect.objectContaining( {
-					type: 'focus',
-				} )
+			await waitFor( () =>
+				expect( screen.getByText( 'tooltip text' ) ).toBeVisible()
 			);
 
 			// Wait for the tooltip element to be positioned (aligned with the button)
```
Comment (brookewp) [inline packages/components/src/tooltip/test/index.js]: This is being tested in a lot of tests. I still need to look into this further, to see why this is the case.

File: packages/components/src/tooltip/test/index.js [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
 import Tooltip from '../';
+import Button from '../../button';
+
 /**
  * WordPress dependencies
  */
 import { TOOLTIP_DELAY } from '../index.js';
 
-jest.useFakeTimers();
+const props = {
```
Comment (brookewp) [inline packages/components/src/tooltip/test/index.js:19]: I think we could add the anchor as a prop to repeat a bit less in each test which was my intention with adding the tooltip text

File: packages/components/src/tooltip/test/index.js [hunk 3]
```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
...
-					<button>Button 1</button>
-					<button>Button 2</button>
+				<Tooltip { ...props }>
+					<Button>Button 1</Button>
+					<Button>Button 2</Button>
 				</Tooltip>
 			);
 
-			const button = screen.getByText( 'Button 1' );
-			act( () => button.focus() );
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js:33]: Did we intentionally remove the focus here without adding an `await user.tab()` or any other suitable replacement?

File: packages/components/src/tooltip/test/index.js [hunk 4]
```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
...
-				<Tooltip text="Help text">
-					<button onFocus={ mockOnFocus }>Hover Me!</button>
+				<Tooltip { ...props }>
+					<Button>Hover Me!</Button>
 				</Tooltip>
 			);
 
-			const button = screen.getByRole( 'button', { name: 'Hover Me!' } );
-			expect( button ).toBeVisible();
+			expect( document.body ).toHaveFocus();
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js]: Is that assertion necessary? I'd expect that the document would be focused by default at the beginning of the test anyway?

File: packages/components/src/tooltip/test/index.js (hunk 1 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: This is because the tooltip takes some time to be displayed and put in the right position. This doesn't happen immediately, so we might want to verify that it has finished before continuing with our tests' assertions.

Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: In this case, though, there are no further assertions, and this is necessary in order to ensure that by the time we're finishing the test, there are no running animations or pending/scheduled updates. IIRC those pending updates are handled by an external library and we don't have much control over them, and they're set in a way that `@testing-library`'s post-test-cleanup doesn't properly handle them, causing bleeding between tests and subsequent test failure to occur.

File: packages/components/src/tooltip/test/index.js [hunk 5]
```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
...
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-
-			act( () => jest.advanceTimersByTime( TOOLTIP_DELAY ) );
-
-			// Tooltip shows after the delay
-			const tooltip = screen.getByText( 'Help text' );
-			expect( tooltip ).toBeVisible();
+			await waitFor( () =>
+				expect( screen.getByText( 'tooltip text' ) ).toBeVisible()
+			);
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js]: Nice 👍 I'd say this (waiting for the tooltip to become visible) is a much better and reliable way for this test to work, rather than relying on a certain fake timer delay.

File: packages/components/src/tooltip/test/index.js [hunk 6]
```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
...
+				screen.getByRole( 'button', { name: /Hover me!/i } )
+			).toHaveFocus();
 
-			// Tooltip is shown after focusing the anchor.
-			const tooltip = screen.getByText( 'Help text' );
-			expect( tooltip ).toBeVisible();
-			expect( mockOnFocus ).toHaveBeenCalledWith(
-				expect.objectContaining( {
-					type: 'focus',
-				} )
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js:73]: This was also testing that our custom `onFocus` was being called. I don't see why it wouldn't be called 🤔 But at the same time, I wanted to be sure that we're intentionally removing this assertion.

File: packages/components/src/tooltip/test/index.js [hunk 7]
```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
...
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-
-			act( () => jest.advanceTimersByTime( TOOLTIP_DELAY ) );
-
-			// Tooltip still hasn't appeared yet, even though the component was focused
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-			expect( mockOnFocus ).toHaveBeenCalledWith(
-				expect.objectContaining( {
-					type: 'focus',
-				} )
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js:144]: Same as above on removing the focus handler assertion.

File: packages/components/src/tooltip/test/index.js [hunk 8]
```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
...
+			await user.click(
+				screen.getByRole( 'button', { text: 'Hover Me!' } )
 			);
+
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js]: How are we certain that the tooltip won't appear a bit after we ran this assertion? The previous version of the test was waiting for some time and doing the check again.

File: packages/components/src/tooltip/test/index.js [hunk 9]
```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
...
 				</Tooltip>
 			);
 
 			const button = screen.getByRole( 'button', { name: 'Hover Me!' } );
-			expect( button ).toBeVisible();
 
 			await user.hover( button );
 
-			// Tooltip hasn't appeared yet
-			expect( mockOnMouseEnter ).toHaveBeenCalledTimes( 1 );
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js:174]: We're removing some more mock call assertions in this test, could you elaborate on the rationale?

File: packages/components/src/tooltip/test/index.js [hunk 10]
```diff
@@ -197,60 +167,44 @@ describe( 'Tooltip', () => {
...
 
-			// Note: this is testing for implementation details,
-			// but couldn't find a better way.
-			const buttonRect = button.getBoundingClientRect();
-			// eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
-			const eventCatcher = container.querySelector( '.event-catcher' );
-			const eventCatcherRect = eventCatcher.getBoundingClientRect();
-			expect( buttonRect ).toEqual( eventCatcherRect );
-
-			await user.hover( eventCatcher );
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js]: I'm not sure the new version of the test reflects what this was doing. Thoughts?

File: packages/components/src/tooltip/test/index.js (hunk 4 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: Fair point! I saw this in an example so I thought it was a good first check to do before proceeding, but if it's redundant then I'm happy to remove it. 😄

File: packages/components/src/tooltip/test/index.js (hunk 8 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: If a button is clicked and the cursor doesn't move, then the tooltip would appear after reaching the set `delay`. But if there was no `delay`, a mouse click would trigger it. My understanding is this test is assuring that an initial mouse click triggers the focus without a tooltip if there is a delay set. 

If that's not the intention, then would it make sense to click and move the cursor off of the button? Or to combine this with test `should not show tooltip if the mouse leaves the anchor before the tooltip has shown` and have the first interaction be a mouse click instead of hover. Thoughts?

File: packages/components/src/tooltip/test/index.js (hunk 9 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js:174]: I thought these were important to test in `should not show tooltip if the mouse leaves the anchor before the tooltip has shown` but not as much here. So I thought these could be simplified to just what needed to be tested, the custom delay.

I'd love to learn why it would be good to have these here as well, but then not in every test (i.e. the test `should render the tooltip when anchor is hovered`).

File: packages/components/src/tooltip/test/index.js (hunk 10 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: I maybe be misunderstanding, or perhaps the test is mislabelled and an additional test is needed? When I see the text `should show tooltip when an element is disabled`, I assumed that is all that needs to be checked. 

And with the new tooltip, we won't have the 'event catcher' so then the test wouldn't translate to the new one. So maybe it would be best to separate this and then leave the implementation detail for a separate test that won't be used for the new tooltip?

File: packages/components/src/tooltip/test/index.js (hunk 6 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js:73]: I did not realize we had a custom `onFocus`. So I thought asserting that the button was in focus was all that was necessary. The event handlers in this version of tooltip are a bit confusing to me, to be honest. 

So I'm curious, what would be a scenario where asserting that the button`.toHaveFocus()` wouldn't find that the mock focus would?

File: packages/components/src/tooltip/test/index.js (hunk 1 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: Thanks for sharing and for your initial comments on this draft! 🙇‍♀️

File: packages/components/src/tooltip/test/index.js (hunk 3 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js:33]: tl;dr, it was somewhat intentionally unintentional. 😅 But in this case, it does make sense to ensure the button has focus through some `userEvent` method, so I will make that change. 

I had removed it [after reading this](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role#accessibility_concerns):
>If the information is important enough for a tooltip, isn't it important enough to always be visible? 

So I was just thinking about if the tooltip should render (when there is just one child) even when the anchor isn't in focus but stay hidden until its conditions are met. I think in most cases, it's just a visual duplicate of an aria-label, so it shouldn't be needed in the DOM. But then our Storybook example shows a default use case where the tooltip has more information. And in this test, the original tooltip text was called 'help text' which is why I changed it to 'tooltip text' because I think it could be misleading to refer to it as help text or additional info etc.

So the way it renders likely shouldn't change, but maybe the examples and documentation to make sure the use case for a tooltip is clear.

File: packages/components/src/tooltip/test/index.js [hunk 11]
```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
 	describe( '#render()', () => {
```
Comment (brookewp) [inline packages/components/src/tooltip/test/index.js]: As a last step, I think we could remove the nested describe here

File: packages/components/src/tooltip/test/index.js (hunk 4 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: Yeah, AFAIK `document.activeElement` will default to `body` in browsers, but also in jsdom:

https://github.com/jsdom/jsdom/blob/04f6c13f4a4d387c7fc979b8f62c6f68d8a0c639/lib/jsdom/living/nodes/DocumentOrShadowRoot-impl.js#L8

File: packages/components/src/tooltip/test/index.js (hunk 6 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js:73]: I took some more time and can't see a good reason to do it the way it was done before. Asserting that the element has the focus is a better way to do this 👍

File: packages/components/src/tooltip/test/index.js (hunk 8 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: I think we're talking about different things.

Previously, the test was waiting for some time before checking if the tooltip was there or not. Now, we're not waiting for any time - we just perform the assertion after the click. So because we didn't wait enough, we can't be sure that after the assertion, the tooltip won't appear. 

Also, I get your point about the `delay` but I don't see us setting a delay for this test, and AFAIK it's `0` by default? 

Let me know if I still don't make sense.

File: packages/components/src/tooltip/test/index.js (hunk 9 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js:174]: Yup, I think you're right. We can actually claim that these previous assertions with mocks were testing implementation details.

File: packages/components/src/tooltip/test/index.js (hunk 10 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: What you're suggesting makes sense, especially in the context of the new tooltip that doesn't have the event catcher.

Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: To clarify, do you mean the suggestion of separating the test into two: one for checking a disabled element and a new one to test event handling? 

Or the original change?

File: packages/components/src/tooltip/test/index.js (hunk 6 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js:73]: Thanks for taking a second look! ❤️

File: packages/components/src/tooltip/test/index.js (hunk 8 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: By default, the delay is set to 700ms. I don't think there are many cases where we'd want a delay of 0 for a tooltip because then it would show up on a mouse click or a swift movement across the element 😄 

Would you find it preferable if we added a timeout to the assertion? i.e.

```js
await waitFor(
	() =>
		expect(
			screen.queryByText( 'tooltip text' )
		).not.toBeInTheDocument(),
	{ timeout: TOOLTIP_DELAY }
);
```

Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: I might be missing something, but isn't the `delay` set to `0` by default? That's what the documentation indicates too:

https://testing-library.com/docs/user-event/options/#delay

Unless we alter it when we're calling `userEvent.setup()`, which we don't seem to be doing in this test.

File: packages/components/src/tooltip/test/index.js (hunk 10 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: I meant the original change. Feel free to do as you prefer with regards to separating the change. If we're getting rid of this version of the component soon, it might make sense to have 2 separate tests and then we'd just remove the event catcher one with the new version of the component.

File: packages/components/src/tooltip/test/index.js (hunk 8 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: Ah, we are talking about two different delays! I'm referring to the `Tooltip` prop and now I see you are referring to `userEvent` options. 🙂 

I'm curious why adding a timeout wouldn't have the same effect?

Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: I'm not saying it won't, was just trying to clarify. To resolve the primary topic here: I think some additional timeout in a `waitFor()` as you suggested will do the job as well 👍

File: packages/components/src/tooltip/test/index.js (hunk 3 above)
Comment (ciampo) [reply packages/components/src/tooltip/test/index.js:33]: This is an interesting point. From the [WAI-ARIA specs](https://w3c.github.io/aria-practices/#tooltip):

> A tooltip is a popup that displays information related to an element when the element receives keyboard focus or the mouse hovers over it. It typically appears after a small delay and disappears when Escape is pressed or on mouse out.

The use of "appears"/"disappears" does not make it clear whether a tooltip should be always rendered (but visually hidden), or whether it should not be rendered at all when not triggered.

I guess we can keep the current behaviour / testing, and see if this feat is supported in `ariakit` or if it's worth implementing it

File: packages/components/src/tooltip/test/index.js (hunk 1 above)
Comment (ciampo) [reply packages/components/src/tooltip/test/index.js]: I'm curious to know: could we simply test for the `tooltip text` `.toBeVisible()`, or is the `toBePositionedPopover()` check necessary to make sure that the animation actually completed?

File: packages/components/src/tooltip/test/index.js [hunk 12]
```diff
@@ -1,193 +1,165 @@
...
+					)
 				).toBePositionedPopover()
 			);
 		} );
 
-		it( 'should render children with additional tooltip when hovered', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
+		it( 'should render the tooltip when anchor is hovered', async () => {
```
Comment (ciampo) [inline packages/components/src/tooltip/test/index.js]: The term "anchor" may be confused with the `<a />` HTML element (the anchor tag), could we use the term "tooltip trigger" instead ?

This may sound super nitty, but it actually got me quite confused for a while, since for a second I thought that if the trigger is a `<a />` element, then a mouseover would suffice 😂

File: packages/components/src/tooltip/test/index.js [hunk 13]
```diff
@@ -1,193 +1,165 @@
...
-
-			// Tooltip hasn't appeared yet after the usual delay
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
 
-			// Advance time again, so that we reach the full TEST_DELAY time
-			act( () => jest.advanceTimersByTime( TEST_DELAY - TOOLTIP_DELAY ) );
+			const tooltip = await screen.findByText( 'tooltip text' );
```
Comment (ciampo) [inline packages/components/src/tooltip/test/index.js]: Is there a reason why you chose to use `screen.findByText` here, instead of `waitFor` ? Or is it just personal preference?

I don't have a strong opinion, I'm just curious :)

File: packages/components/src/tooltip/test/index.js [hunk 14]
```diff
@@ -261,111 +217,131 @@ describe( 'Tooltip', () => {
...
 
 			// Advance time by MOUSE_LEAVE_DELAY time
-			act( () => jest.advanceTimersByTime( MOUSE_LEAVE_DELAY ) );
+			await waitFor(
+				() =>
+					expect(
+						screen.queryByText( 'tooltip text' )
+					).not.toBeInTheDocument(),
+				{ timeout: MOUSE_LEAVE_DELAY }
+			);
```
Comment (ciampo) [inline packages/components/src/tooltip/test/index.js]: I wonder if this assertion is actually testing what we want.

Since the tooltip shouldn't be visible , this `waitFor` call would possibly immediately "pass", instead of performing the check at the end of the timeout time.

Basically, using `waitFor` works well when waiting for a condition to happen asynchronously — but it's likely not the right tool to test for a condition NOT to change for a specific amount of time.

@tyxla , do you have any thoughts? If my assumptions are correct, we may have to rewrite a few assertions throughout this file.

**Review by ciampo — COMMENTED** [review]
Great work so far, @brookewp !

These tests are looking much better, and will serve us well when migrating to the new version of `Tooltip`

File: packages/components/src/tooltip/test/index.js (hunk 14 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: Yep, I've taken a look, and it seems like `timeout` doesn't work the way the current version of the assertion assumes.

The timeout works like this: it will run on every X ms (X being the specified `interval`, 50ms by default), and if the assertion passes on the first run, it will resolve.

That means that we should instead actually set a timeout, and perform an assertion after that timeout.

Alternatively, we could use [`waitForElementToBeRemoved()`](https://testing-library.com/docs/dom-testing-library/api-async/#waitforelementtoberemoved), which should better express what we're aiming to do here, without having to specifically play with setting up intervals and timeouts.

File: packages/components/src/tooltip/test/index.js (hunk 12 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: Good point!

File: packages/components/src/tooltip/test/index.js (hunk 14 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: Thank you both! That makes sense why `waitFor` doesn't work well in this situation, where we are testing that something shouldn't happen after a set time. 

However, I am unsure if I understand why `waitForElementToBeRemoved()` would work in this case. The tooltip should never be rendered because the mouse moved away from the anchor before it's set `delay`. So how would waiting for it to be removed from the DOM work here?

File: packages/components/src/tooltip/test/index.js (hunk 13 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: I think it was where I started, and it was working better, but I've since updated this to be `waitFor` as well so that'll be in my next commit. 🙂

File: packages/components/src/tooltip/test/index.js (hunk 1 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: From @tyxla's explanation above I assumed that `.toBeVisible()` wasn't enough to ensure the animations/updates have resolved, unless I'm misunderstanding?

Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: What you said, Brooke, that's my understanding as well.

File: packages/components/src/tooltip/test/index.js (hunk 14 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: Yep, you're right, `waitForElementToBeRemoved()` won't do the job, because it's already removed and will immediately pass.

I guess we can do the following then:

1. Add a manual timeout:

```
await new Promise( ( resolve ) => setTimeout( resolve, MOUSE_LEAVE_DELAY ) );
```

2. Add the `.not.toBeInTheDocument()` assertion afterwards. 

How does that sound?

Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: Sounds great, thank you!

File: packages/components/src/tooltip/test/index.js [hunk 15]
```diff
@@ -1,371 +1,357 @@
...
 			const eventCatcher = container.querySelector( '.event-catcher' );
 			await user.click( eventCatcher );
 			expect( onClickMock ).not.toHaveBeenCalled();
-		} );
 
-		it( 'should not show tooltip if the mouse leaves the anchor before the tooltip has shown', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
+			const eventCatcherRect = eventCatcher.getBoundingClientRect();
```
Comment (brookewp) [inline packages/components/src/tooltip/test/index.js]: Instead of creating another test [as we discussed](https://github.com/WordPress/gutenberg/pull/48397/files#r1121621557), I moved the implementation details to this test which is also testing the event catcher for a disabled button. Thoughts?

File: packages/components/src/tooltip/test/index.js [hunk 16]
```diff
@@ -1,371 +1,357 @@
...
+			const user = userEvent.setup();
 
 			render(
 				<Tooltip
-					text="Help text"
+					{ ...props }
 					shortcut={ {
-						display: 'shortcut text',
-						ariaLabel: 'shortcut label',
+						display: '⇧⌘,',
```
Comment (brookewp) [inline packages/components/src/tooltip/test/index.js]: I changed the values of the shortcut object to test the shortcut's purpose, which is to render keyboard shortcuts and corresponding aria-labels for screenreaders.

File: packages/components/src/tooltip/test/index.js (hunk 12 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: I updated to tooltip trigger, but what do you think of 'tooltip anchor'? 😂 In ariakit, they also use 'anchor' for their tooltip (`TooltipAnchor`) so I think it would be nice to keep that language.

File: packages/components/src/tooltip/test/index.js (hunk 15 above)
Comment (tyxla) [reply packages/components/src/tooltip/test/index.js]: Sounds good to me, thanks 👍

File: packages/components/src/tooltip/test/index.js (hunk 12 above)
Comment (ciampo) [reply packages/components/src/tooltip/test/index.js]: That also works!

File: packages/components/src/tooltip/test/index.js [hunk 17]
```diff
@@ -1,373 +1,323 @@
...
+		expect( screen.queryByText( 'tooltip text' ) ).not.toBeInTheDocument();
+		expect( onMouseEnterMock ).toHaveBeenCalledTimes( 1 );
 
-			// Tooltip hasn't appeared yet
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-			expect( onMouseEnterMock ).toHaveBeenCalledTimes( 1 );
+		// Advance time by MOUSE_LEAVE_DELAY time
+		await new Promise( ( resolve ) =>
+			setTimeout( resolve, MOUSE_LEAVE_DELAY )
+		);
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js:242]: This could be defined as a helper function since we use it multiple times. Totally optional, since it's small enough too.

File: packages/components/src/tooltip/test/index.js [hunk 18]
```diff
@@ -1,373 +1,323 @@
...
 /**
  * Internal dependencies
  */
 import Tooltip from '../';
+import Button from '../../button';
+
 /**
  * WordPress dependencies
  */
 import { TOOLTIP_DELAY } from '../index.js';
```
Comment (tyxla) [inline packages/components/src/tooltip/test/index.js]: This looks like an internal dependency, no?

I realize this was here before, but perhaps we could address it while we're here?

**Review by tyxla — APPROVED** [review]
Thank you, great work @brookewp! 🚀 

Tooltip tests are in a much better state now!

🚢

File: packages/components/src/tooltip/test/index.js (hunk 18 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js]: Yes, good catch!

File: packages/components/src/tooltip/test/index.js (hunk 17 above)
Comment (brookewp) [reply packages/components/src/tooltip/test/index.js:242]: Thanks for pointing this out! Since it's just two instances, I'll leave it but it's definitely something I need to be mindful of in the future to avoid too much repetition.
